lib/AST/DiagnosticEngine.cpp

```diff
@@ -311,6 +313,32 @@ void Diagnostic::addChildNote(Diagnostic &&D) {
   ChildNotes.push_back(std::move(D));
 }
 
+YAMLLocalizationProducer::YAMLLocalizationProducer(std::string locale,
+                                                   std::string path) {
+  llvm::SmallString<128> DiagnosticsFilePath(path);
+  llvm::sys::path::append(DiagnosticsFilePath, locale);
+  llvm::sys::path::replace_extension(DiagnosticsFilePath, ".yaml");
+  auto FileBufOrErr = llvm::MemoryBuffer::getFileOrSTDIN(DiagnosticsFilePath);
+  // Resize Diags from YAML file to be the same size
+  // as diagnosticStrings from def files.
+  diagnostics.resize(LocalDiagID::NumDiags);
```

Comment:
To handle the absence of localization files I decided to resize the size of the `diagnostics` here.

Reply:
Why waste memory of having multiple thousand of empty entries in the list if we could simplify check whether it's empty in `getMessageOr`?